targets/TARGET_NORDIC/TARGET_NRF5/us_ticker.c

```diff
@@ -500,74 +507,69 @@ static void register_next_tick() {
  * @note this function shouldn't be called directly.
  * @return  IRQ number of the alternative hardware timer
  */
-int os_tick_init (void)
+int32_t osRtxSysTimerSetup(void)
 {
     common_rtc_init();
+    
+    os_rtc_peiod = (RTC0_CONFIG_FREQUENCY) / osRtxConfig.tick_freq;
+
+    return nrf_drv_get_IRQn(COMMON_RTC_INSTANCE);
+}
+
+// Start SysTickt timer emulation
+void osRtxSysTimerEnable(void)
+{
     nrf_rtc_int_enable(COMMON_RTC_INSTANCE, OS_TICK_INT_MASK);
 
-    nrf_rtc_cc_set(COMMON_RTC_INSTANCE, OS_TICK_CC_CHANNEL, 0);
-    register_next_tick();
+    uint32_t current_cnt = nrf_rtc_counter_get(COMMON_RTC_INSTANCE);
+    nrf_rtc_cc_set(COMMON_RTC_INSTANCE, OS_TICK_CC_CHANNEL, current_cnt);
+    register_next_tick();    
+}
 
-    return nrf_drv_get_IRQn(COMMON_RTC_INSTANCE);
+// Stop SysTickt timer emulation
+void osRtxSysTimerDisable(void)
+{
+    nrf_rtc_int_disable(COMMON_RTC_INSTANCE, OS_TICK_INT_MASK);
 }
 
+
+
 /**
  * Acknowledge the tick interrupt.
  * This function is called by the function OS_Tick_Handler of RTX.
  * @note this function shouldn't be called directly.
  */
-void os_tick_irqack(void)
+void osRtxSysTimerAckIRQ(void)
 {
     clear_tick_interrupt();
     register_next_tick();
 }
 
-/**
- * Returns the overflow flag of the alternative hardware timer.
- * @note This function is exposed by RTX kernel.
- * @return 1 if the timer has overflowed and 0 otherwise.
- */
-uint32_t os_tick_ovf(void) {
-    uint32_t current_counter = nrf_rtc_counter_get(COMMON_RTC_INSTANCE);
-    uint32_t next_tick_cc_value = nrf_rtc_cc_get(COMMON_RTC_INSTANCE, OS_TICK_CC_CHANNEL);
-
-    return is_in_wrapped_range(previous_tick_cc_value, next_tick_cc_value, current_counter) ? 0 : 1;
+// provide a free running incremental value over the entire 32-bit range
+uint32_t osRtxSysTimerGetCount(void) {
```

Comment:
That would make sense, if the timer is supposed to be stopped, it doesn't make sense to return a moving value.